source/Hardpoint.h

```diff
@@ -30,7 +31,13 @@ class Visual;
 class Hardpoint {
 public:
 	// Constructor. Hardpoints may or may not specify what weapon is in them.
-	Hardpoint(const Point &point, const Angle &baseAngle, bool isTurret, bool isParallel, const Outfit *outfit = nullptr);
+	// If isTurret is true, angles have various parameters according to the size:
+	//  0: The turret is omnidirectional and points to the front when holding.
+	//  1: The turret is omnidirectional and points to the angles[0] when holding.
+	//  2: The turret can rotate in the range from angles[0] to angles[1] and points to the center of the range when holding.
+	//  >=3: The turret can rotate in the range from angles[0] to angles[1] and points to the angles[2] when holding.
+	// If isTurret is false, angles[0] specify the angle of firing direction.
+	Hardpoint(const Point &point, const std::vector<Angle> &angles, bool isTurret, bool isParallel, const Outfit *outfit = nullptr);
```

Comment:
I'm not too happy with using a vector here, especially since all 3 angles have a different meaning.
What if we just add a new variable pair<Angle> turnRange?

Reply:
I agree for not using a vector. I hesitated to add a dependence on Hardpoint.h to Armament.h at that time.
It's need to provide a holding position.

Reply:
I'm not sure we need to provide a holding position, we don't do that today either.